### PR TITLE
Map builder

### PR DESCRIPTION
**Pull request by n2oneProgrammer**

### CODE REVIEW

File: tests/map_builder_test.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+        )
+
+        result = self.map_builder.construct_object(map_object)[0]
+        self.assertIsInstance(result, Rectangle)
+        self.assertEqual(result.position, expect.position)
+        self.assertEqual(result.width, expect.width)
+        self.assertEqual(result.height, expect.height)
+        self.assertEqual(result.color, expect.color)
+
+    # STRETCH OPCTIONS
```
Comment (mwawrzen) [inline tests/map_builder_test.py]: In my opinion it should be "OPTIONS" instead of "OPCTIONS" but idk, good night

**Review by mwawrzen — APPROVED** [review]
LGTM

File: models/map_builder.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                    result_list.append(self._construct_1_object(object_json, [x_stretch,
+                                                                              y_stretch]))
+            return result_list
+
+        if "position" not in object_json:
+            raise ValueError(f"Object {object_json['type']} must have position")
+        return [self._construct_1_object(object_json, object_json['position'])]
+
+    def _construct_1_object(self, object_json, position: list):
+        if object_json["type"] == "sprite":
```
Comment (jaanonim) [inline models/map_builder.py]: Use `method = getattr(self,name,default_method)`

**Review by jaanonim — COMMENTED** [review]
One comment

File: models/map_builder.py (hunk 2 above)
Comment (n2oneProgrammer) [reply models/map_builder.py]: Is good idea but in this place I know the attribute "type" exist

**Review by Adron4444 — APPROVED** [review]
LGTM

**Pull request merged**